pkg/standalone-utils/test/VaultActions.test.ts

```diff
@@ -1161,4 +1190,324 @@ describe('VaultActions', function () {
       });
     });
   });
+
+  describe('user balance ops', () => {
+    const amountDAI = fp(2);
+    const amountSNX = fp(5);
+
+    context('when caller is not authorized', () => {
+      it('reverts', async () => {
+        await expect(
+          relayer.connect(other).multicall([
+            encodeManageUserBalance({
+              ops: [
+                {
+                  kind: UserBalanceOpKind.DepositInternal,
+                  asset: tokens.DAI.address,
+                  amount: amountDAI,
+                  sender: user.address,
+                },
+              ],
+            }),
+          ])
+        ).to.be.revertedWith('Incorrect sender');
+      });
+    });
+
+    context('when caller is authorized', () => {
+      let sender: Account;
+
+      context('sender = user', () => {
+        beforeEach(() => {
+          sender = user;
+        });
+
+        itTestsUserBalance();
+      });
+
+      context('sender = relayer', () => {
+        sharedBeforeEach('fund relayer with tokens and approve vault', async () => {
+          sender = relayer;
+          await tokens.DAI.transfer(relayer, amountDAI, { from: user });
+          await tokens.SNX.transfer(relayer, amountSNX, { from: user });
+          await approveVaultForRelayer(relayerLibrary, user, tokens);
+        });
+
+        itTestsUserBalance();
+      });
+
+      function itTestsUserBalance() {
+        it('sends immediate amounts', async () => {
+          // Internal balance of sender doesn't change
+          // Tokens are transferred from sender to recipient's internal balance
+
+          await expectBalanceChange(
```

Comment:
would it make sense to track the external balance changes..?
Just thinking out loud here, it might be an overkill. This is just a thought because without considering the tiny detail that by sending the vault we are tracking internal balances the changes are not intuitive (i.e. "why is the sender balance not changing?" was the first thing that came to my mind).

Reply:
Well, I wrote it this way because the function under test specifically affects internal balances. I added a big documentation block explaining how it works, so hopefully that will address the concern.